### PR TITLE
fix:don't prompt user for multisig config if chain is core

### PR DESCRIPTION
**Pull request by aroralanuk**
### Description

- don't restrict user to having two chains for ism config
- if the user accidentally picks two chains, we prompt them again to confirm if they don't want to use the hyperlane validators for their multisigconfig

### Drive-by changes

None

### Related issues

- fixes https://github.com/hyperlane-xyz/issues/issues/811

### Backward compatibility

Yes

### Testing

Manual between arbitrumgoerli and anvil1

### CODE REVIEW

File: typescript/cli/src/config/multisig.ts [hunk 1]
```diff
@@ -84,6 +92,19 @@ export async function createMultisigConfig({
       result[chain] = lastConfig;
       continue;
     }
+    if (Object.keys(sdkContractAddressesMap).includes(chain)) {
+      const reuseCoreConfig = await confirm({
+        message: 'Use existing Hyperlane validators for this chain?',
+      });
+      if (reuseCoreConfig) {
+        result[chain] = {
```
Comment (nambrot) [inline typescript/cli/src/config/multisig.ts:100]: Should we just leave it out of `result` in this case? We can still merge it during the deployment step, but allows the SDK to be the source of truth

File: typescript/cli/src/config/multisig.ts [hunk 2]
```diff
@@ -84,6 +92,19 @@ export async function createMultisigConfig({
...
+          threshold: defaultMultisigConfigs[chain].threshold,
+          validators: defaultMultisigConfigs[chain].validators,
+        };
+        console.log(result[chain]);
```
Comment (nambrot) [inline typescript/cli/src/config/multisig.ts:104]: Remove

File: typescript/cli/src/utils/chains.ts [hunk 3]
```diff
@@ -33,9 +32,9 @@ export async function runSingleChainSelectionStep(
 export async function runMultiChainSelectionStep(
   customChains: ChainMap<ChainMetadata>,
   message = 'Select chains',
-  chainsToFilterOut: ChainName[] = [],
+  chainSelection = false,
```
Comment (nambrot) [inline typescript/cli/src/utils/chains.ts:35]: chainSelection is not descriptive really. How about `requireMultiple` or something like that? Also, wha

File: typescript/cli/src/utils/chains.ts [hunk 4]
```diff
@@ -44,19 +43,14 @@ export async function runMultiChainSelectionStep(
       pageSize: 20,
     })) as string[];
     handleNewChain(chains);
-    if (chains?.length >= 2) return chains;
+    if (chains?.length >= 2 || !chainSelection) return chains;
```
Comment (nambrot) [inline typescript/cli/src/utils/chains.ts:46]: Nit: I personally like reading these conditions in the reverse, basically `if(badCondition) error`

and then just return chains if all none of the bad conditions triggered

File: .changeset/ninety-seas-chew.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+'@hyperlane-xyz/cli': patch
+---
+
+Allow users to config their own PI chain
```
Comment (nambrot) [inline .changeset/ninety-seas-chew.md:5]: More descriptive message here please. Something like `Allow users to only configure validators for their chain`

File: typescript/cli/src/config/multisig.ts (hunk 1 above)
Comment (aroralanuk) [reply typescript/cli/src/config/multisig.ts:100]: Makes sense. I was thinking if the user selected the chain, they'll expect to see the config in their yaml file. But. I'm don't feel strongly about either approach.

File: typescript/cli/src/utils/chains.ts (hunk 3 above)
Comment (aroralanuk) [reply typescript/cli/src/utils/chains.ts:35]: 1) What

Comment (nambrot) [reply typescript/cli/src/utils/chains.ts]: sorry mid-sentence i just asked you in person what `chainsToFilterOut` was used for

**Review by nambrot — APPROVED** [review]

**Pull request merged**